gradle-plugin/src/functionalTest/groovy/com/google/cloud/tools/dependencies/gradle/BuildStatusFunctionalTest.groovy

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.dependencies.gradle
+
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+import static org.gradle.testkit.runner.TaskOutcome.FAILED
+import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
+
+class BuildStatusFunctionalTest extends Specification {
+  @Rule TemporaryFolder testProjectDir = new TemporaryFolder()
+  File buildFile
+
+  def setup() {
+    buildFile = testProjectDir.newFile('build.gradle')
+    buildFile << """
+        plugins {
+            id 'java'
+            id 'com.google.cloud.tools.linkagechecker'
+        }
+        """
+  }
+
+  def "can validate a project with no dependency conflicts"() {
+    buildFile << """
+        repositories {
+          mavenCentral()
+        }
+        
+        // Guava does not have any linkage error
+        dependencies {
+          compile 'com.google.guava:guava:28.2-jre'
+        }
+        
+        linkageChecker {
+          configurations = ['compile']
+        }
+        """
+
+    when:
```

Comment:
These "when" and "then" come from Spock test framework http://spockframework.org/spock/docs/1.3/spock_primer.html#_when_and_then_blocks (recommended in the Gradle test documentation)

Reply:
indentation seems off, but I don't know Groovy conventions

Reply:
Spock's blocks are not indented. The Gradle Plugin documentation example's UrlVerifierPluginFunctionalTest.groovy in https://guides.gradle.org/testing-gradle-plugins/#functional-tests as well as Spock's example code (http://spockframework.org/spock/docs/1.3/spock_primer.html#_when_and_then_blocks) have this style too.